source/extensions/tracers/datadog/datadog_tracer_impl.cc

```diff
@@ -0,0 +1,129 @@
+#include "extensions/tracers/datadog/datadog_tracer_impl.h"
+
+#include "common/common/enum_to_int.h"
+#include "common/common/fmt.h"
+#include "common/common/utility.h"
+#include "common/http/headers.h"
+#include "common/http/message_impl.h"
+#include "common/http/utility.h"
+#include "common/tracing/http_tracer_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+
+Driver::TlsTracer::TlsTracer(const std::shared_ptr<opentracing::Tracer>& tracer,
+                             TraceReporterPtr&& reporter, Driver& driver)
+    : tracer_(tracer), reporter_(std::move(reporter)), driver_(driver) {}
+
+Driver::Driver(const envoy::config::trace::v2::DatadogConfig& datadog_config,
+               Upstream::ClusterManager& cluster_manager, Stats::Store& stats,
+               ThreadLocal::SlotAllocator& tls, Runtime::Loader& runtime)
+    : OpenTracingDriver{stats},
+      cm_(cluster_manager), tracer_stats_{DATADOG_TRACER_STATS(
+                                POOL_COUNTER_PREFIX(stats, "tracing.datadog."))},
+      tls_(tls.allocateSlot()), runtime_(runtime) {
+
+  Upstream::ThreadLocalCluster* cluster = cm_.get(datadog_config.collector_cluster());
```

Comment:
You can use `Config::Utility::checkCluster` here I think.

Reply:
Sure please clean up.

Reply:
Done for all three cases.